components/org.wso2.carbon.identity.api.server.tenant.management/org.wso2.carbon.identity.api.server.tenant.management.v1/src/main/java/org/wso2/carbon/identity/api/server/tenant/management/v1/core/ServerTenantManagementService.java

```diff
@@ -196,6 +197,24 @@ private Tenant createTenantInfoBean(TenantModel tenantModel) throws TenantManage
             tenant.setAdminFirstName(tenantModel.getOwners().get(0).getFirstname());
             tenant.setAdminLastName(tenantModel.getOwners().get(0).getLastname());
             tenant.setEmail(tenantModel.getOwners().get(0).getEmail());
+            String provisioningMethod = tenantModel.getOwners().get(0).getProvisioningMethod();
+            if (INLINE.equalsIgnoreCase(provisioningMethod) || "invite-via-email".
+                    equalsIgnoreCase(provisioningMethod)) {
+                if (INLINE.equalsIgnoreCase(provisioningMethod)) {
+                    String password = tenantModel.getOwners().get(0).getPassword();
+                    if (StringUtils.isBlank(password)) {
+                        throw new TenantManagementClientException(TenantConstants.ErrorMessage.
+                                ERROR_CODE_MISSING_REQUIRED_PARAMETER.getCode(),
+                                String.format(TenantConstants.ErrorMessage.
+                                        ERROR_CODE_MISSING_REQUIRED_PARAMETER.getMessage(), "password"));
+                    }
+                    tenant.setAdminPassword(password);
+                }
+                tenant.setProvisioningMethod(provisioningMethod);
+            } else {
+                throw new TenantManagementClientException(TenantConstants.ErrorMessage.
```

Comment:
provisioningMethod should be extensible. We better not limit to INLINE and invite-via-email. We don't need to check this and throw an error. we can write customizations based on the parameter value.